Implemented Equal(Object) and GetHashCode() methods for domain entities

ky-lian opened the pull request:
Also, appropriate tests were added.

fvxrhj committed:
Implemented Equal(Object) and GetHashCode() methods for domain entities

Also, appropriate tests were added.

alex-ks commented on Comptech.Backend/src/Comptech.Backend.Data/DomainEntities/Photo.cs:
```diff
@@ -18,5 +18,24 @@ public Photo(int sessionID, byte[] image, DateTime timeStamp)
+            {
+                return false;
+            }
+            else
```
Else после блока, который заканчивается на return - лишний. Без else вложенность остального кода уменьшится на 1.

ky-lian replied:
Понял, сейчас везде поправлю

alex-ks commented on Comptech.Backend/src/Comptech.Backend.Data/DomainEntities/Point.cs:
```diff
@@ -6,14 +6,32 @@
+            {
+                return false;
+            }
+            else
```
См. комментарий выше.

alex-ks commented on Comptech.Backend/src/Comptech.Backend.Data/DomainEntities/Points.cs:
```diff
@@ -16,6 +16,24 @@ public Points(Point topLeft, Point bottomRight)
+            {
+                return false;
+            }
+            else
```
См. комментарий выше.

alex-ks commented on Comptech.Backend/src/Comptech.Backend.Data/DomainEntities/Pulse.cs:
```diff
@@ -17,6 +17,25 @@ public Pulse(int sessionID, int bpm, DateTime timeStamp)
+            {
+                return false;
+            }
+            else
```
См. комментарий выше.

alex-ks commented on Comptech.Backend/src/Comptech.Backend.Data/DomainEntities/RecognitionResults.cs:
```diff
@@ -17,6 +17,25 @@ public RecognitionResults(bool isValid, Points coords, int photoID)
+            {
+                return false;
+            }
+            else
```
См. комментарий выше.

alex-ks commented on Comptech.Backend/src/Comptech.Backend.Data/DomainEntities/Session.cs:
```diff
@@ -10,16 +10,35 @@ public class Session
+            {
+                return false;
+            }
+            else
```
См. комментарий выше.

fvxrhj committed:
Removed excess "else" statements in Equals(Object) methods of domain entities classes.

Also, added new lines in test classes (for correct work of diff) and removed obsolete debug code from GenericEqualityTest.TestGetHashCode method.
